### PR TITLE
Improved syntax denfinition for dashboard syntaxes

### PR DESCRIPTION
**Pull request by aziz**
## Status Syntax 
- Added more granular scopes to be able to make better custom color schemes
- No change for non customized color shcemes
- Added test for status syntax

## Branch Syntax 
- More scopes for branch syntax
- No change for non customized color shcemes
- Extractted shared syntax to dashboard.sublime-syntax
- Added test for dashboard syntax

## Tags Syntax
- More detailed scopes 
- No change for non customized color shcemes
- Uses the base dashboards syntax
- Added test

## Rebase Syntax
- Uses the base dashboards syntax

screen shot 2016-06-21 at 11 45 43 pm
screen shot 2016-06-22 at 1 08 36 am

### CODE REVIEW

**Comment by divmain** [comment]
Very nice!  I'll want to pull down this branch to confirm there are no breaking changes - if all looks good, expect this to be merged sometime tomorrow.  Thanks again!

**Comment by aziz** [comment]
I've got another branch for improving Rebase dashboard waiting for this one to be merged. 
https://github.com/aziz/GitSavvy/tree/status-rebase-improvments

Here is how it looks:
screen shot 2016-06-23 at 9 40 37 pm

**Comment by stoivo** [comment]
Hello, I am looking at you pull request. All so far look good, I have some question git I will wait until I have had time to look at the hole PR. I haven't had the time so far. I am on a conference now, and will be rest of the weekend.

**Comment by divmain** [comment]
Took me a couple of days, but finally pulled it down.  This looks good - thanks again!

**Comment by stoivo** [comment]
I know the PR is merged but I have  two questions.

File: syntax/dashboard.sublime-syntax [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      captures:
+        1: punctuation.definition.git-savvy.section
+      push:
+        - meta_scope: comment.git-savvy.status.key-bindings-menu
+
+        - match: ^-\n$
+          scope: comment.git-savvy.key-bindings-menu.end
+          pop: true
+
+        - match: "#{3,}"
```
Comment (stoivo) [inline syntax/dashboard.sublime-syntax]: It looks to me like we don't need this section? Do you know if we do?

File: syntax/branch.sublime-syntax [hunk 2]
```diff
@@ -6,64 +6,39 @@ hidden: true
...
+        1: meta.git-savvy.branches.branch.sha1
+        2: meta.git-savvy.branches.branch.name gitsavvy.gotosymbol
+        3: comment.git-savvy.branches.branch.extra-info
+
+    - match: '^  (▸) ([0-9a-f]{7,40}) ([^ ]+)(.*)$'
+      captures:
+        0: meta.git-savvy.branches.branch
+        1: string.other.git-savvy.branches.active-branch.symbol
+        2: string.other.git-savvy.branches.active-branch.sha1
+        3: string.other.git-savvy.branches.active-branch.name. gitsavvy.gotosymbol
```
Comment (stoivo) [inline syntax/branch.sublime-syntax]: Is the dot in the end intended?

**Comment by stoivo** [comment]
I am happy to se your PR for Rebase dashboard :)

File: syntax/dashboard.sublime-syntax (hunk 1 above)
Comment (aziz) [inline syntax/dashboard.sublime-syntax]: This is the part where we mark the help menu section and the `#`s around the help headers. it's required in almost all the dashboards.

File: syntax/branch.sublime-syntax (hunk 2 above)
Comment (aziz) [inline syntax/branch.sublime-syntax]: This is typo. I'll fix it asap.

File: syntax/dashboard.sublime-syntax (hunk 1 above)
Comment (aziz) [inline syntax/dashboard.sublime-syntax]: I got your point now. It might now be needed. I'll investigate this.

Comment (aziz) [inline syntax/dashboard.sublime-syntax]: Here are two screenshots to show which `#`s are matched by this line

screen shot 2016-06-26 at 6 32 06 pm

screen shot 2016-06-26 at 6 32 17 pm

File: syntax/branch.sublime-syntax (hunk 2 above)
Comment (stoivo) [inline syntax/branch.sublime-syntax]: Hehe, I didn't see the difference since it is the same colour with and without.

**Pull request merged**